data/loginUsers.js

```diff
@@ -0,0 +1,78 @@
+const mongoCollections = require("../config/mongoCollections");
+let { ObjectId } = require("mongodb");
+const users = mongoCollections.users;
+const bcrypt = require("bcryptjs");
+const login_validations = require("../validations/login_validations");
+
+let count = 10;
+
+async function checkInput(username, password) {
+  if (!username || !password)
+    throw { code: 400, error: `Username or Password cannot be empty` };
+  if (typeof username != "string")
+    throw { code: 400, error: `Username needs to be a string` };
+  let myUsername = username.split(" ");
+  if (myUsername.length > 1)
+    throw { code: 400, error: `Username cannot have spaces` };
+  if (/[^\w\s]/.test(username))
```

Comment:
I think this regex should just be /[^\w]/ You're testing if the username contains any character that's not \w - (a-z,A-Z,0-9 or _). A username shouldn't have spaces so \s shouldn't be there in the test regex